domain/model/modelmigration/export.go

```diff
@@ -29,26 +29,66 @@ type ExportService interface {
 	// GetEnvironVersion retrieves the version of the environment provider
 	// associated with the model.
 	GetEnvironVersion(context.Context) (int, error)
+
+	// GetModelConstraints returns the currently set constraints for the model.
+	// The following error types can be expected:
+	// - [modelerrors.NotFound]: when no model exists to set constraints for.
+	GetModelConstraints(context.Context) (coreconstraints.Value, error)
 }
 
-// exportOperation describes a way to execute a migration for
-// exporting model.
 type exportOperation struct {
 	modelmigration.BaseOperation
 
-	serviceGetter func(modelUUID model.UUID) ExportService
+	serviceGetter func(modelUUID coremodel.UUID) ExportService
 	logger        logger.Logger
 }
 
+// exportEnvironVersionOperation describes a way to execute a migration for
+// exporting model.
+type exportEnvironVersionOperation struct {
+	exportOperation
+}
+
+// exportModelConstraintsOperation describes an export operation for a model's
+// set constraints.
+type exportModelConstraintsOperation struct {
+	exportOperation
+}
+
 // Name returns the name of this operation.
-func (e *exportOperation) Name() string {
-	return "export model"
+func (e *exportEnvironVersionOperation) Name() string {
+	return "export model environ version"
+}
+
+// Name returns the name of this operation.
+func (e *exportModelConstraintsOperation) Name() string {
+	return "export model constraints"
+}
+
+// newExportEnvironVersionOperation constructs a new
+// [exportEnvironVersionOperation]
+func newExportEnvironVersionOperation(l logger.Logger) *exportEnvironVersionOperation {
+	return &exportEnvironVersionOperation{
+		exportOperation{
+			logger: l,
+		},
+	}
+}
+
+// newExportModelConstraintsOperation constructs a new
+// [exportModelConstraintsOperation]
+func newExportModelConstraintsOperation(l logger.Logger) *exportModelConstraintsOperation {
+	return &exportModelConstraintsOperation{
+		exportOperation{
+			logger: l,
+		},
+	}
 }
 
-// Setup the export operation, this will ensure the service is created
-// and ready to be used.
+// Setup established the required services needed for retrieving information
+// about the model being exported.
 func (e *exportOperation) Setup(scope modelmigration.Scope) error {
-	e.serviceGetter = func(modelUUID model.UUID) ExportService {
+	e.serviceGetter = func(modelUUID coremodel.UUID) ExportService {
```

Comment:
Not sure it's worth keeping `exportOperation` for just reusing this Setup() func.

Reply:
It is worth keeping for the moment because of the complexity introduced by the service getter. This will go away very soon and we can do remove the common operation then.